src/util/datablock/datablock.c

```diff
@@ -13,23 +13,28 @@
 #include <stdbool.h>
 
 // Computes the number of blocks required to accommodate n items.
-#define ITEM_COUNT_TO_BLOCK_COUNT(n) \
-    ceil((double)n / DATABLOCK_BLOCK_CAP)
+#define ITEM_COUNT_TO_BLOCK_COUNT(n, cap) \
```

Comment:
why need cap? can't we just write dataBlock->blockCap?

Reply:
this macro answers the question:
given `n` items how many blocks do we need